### PR TITLE
APPMAN-876 added ImportAudit table

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

**Review by cofaulco — APPROVED** [review]

**Review by cofaulco — APPROVED** [review]

File: src/SFA.DAS.PublicSectorOrganisations.Api/SFA.DAS.PublicSectorOrganisations.Api.csproj [hunk 1]
```diff
@@ -7,23 +7,38 @@
         <InvariantGlobalization>false</InvariantGlobalization>
     </PropertyGroup>
 
+    <ItemGroup>
+      <Content Remove="appsettings.Development.json" />
+      <Content Remove="appsettings.json" />
+    </ItemGroup>
```
Comment (cofaulco) [inline src/SFA.DAS.PublicSectorOrganisations.Api/SFA.DAS.PublicSectorOrganisations.Api.csproj]: can these be removed?

**Review by cofaulco — CHANGES_REQUESTED** [review]

**Review by cofaulco — DISMISSED** [review]

File: src/SFA.DAS.PublicSectorOrganisations.Data.Tests/Ons/OnsDownloadService/WhenCreatingExcelFileFromDownload.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        dateTimeProviderMock.Setup(x => x.UtcNow).Returns(testDate);
+
+        var onsDownloadClientMock = new Mock<IOnsDownloadClient>();
+        onsDownloadClientMock.Setup(x => x.GetAsync(It.IsAny<string>())).ReturnsAsync(getNotFoundResponse);
+
+        try
+        {
+            var sut = new Data.Ons.OnsDownloadService(onsDownloadClientMock.Object, dateTimeProviderMock.Object, config,
+                Mock.Of<ILogger<Data.Ons.OnsDownloadService>>());
+            await sut.CreateLatestOnsExcelFile();
```
Comment (cofaulco) [inline src/SFA.DAS.PublicSectorOrganisations.Data.Tests/Ons/OnsDownloadService/WhenCreatingExcelFileFromDownload.cs]: can we use ShouldThrow() here?

File: src/SFA.DAS.PublicSectorOrganisations.Data.Tests/Ons/OnsExcelReaderServiceUsingClosedXml/WhenReadingAnExcelFile.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        var path = TestContext.CurrentContext.TestDirectory;
+
+        var sut = new Data.Ons.OnsExcelReaderServiceUsingClosedXml(
+            Mock.Of<ILogger<Data.Ons.OnsExcelReaderServiceUsingClosedXml>>());
+
+        try
+        {
+            sut.GetOnsDataFromSpreadsheet(path + @"\Data\pscgjan2024Test.xlsx");
+            Assert.Fail("Should reach here");
```
Comment (cofaulco) [inline src/SFA.DAS.PublicSectorOrganisations.Data.Tests/Ons/OnsExcelReaderServiceUsingClosedXml/WhenReadingAnExcelFile.cs]: can we use ShouldThrow() here?

File: src/SFA.DAS.PublicSectorOrganisations.Data/Ons/OnsExcelReaderService.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
+﻿using SFA.DAS.PublicSectorOrganisations.Domain.Interfaces;
+using SFA.DAS.PublicSectorOrganisations.Domain.PoliceApiClientResponse;
+using System.Data.OleDb;
+using System.Data;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.PublicSectorOrganisations.Domain.Exceptions;
+
+namespace SFA.DAS.PublicSectorOrganisations.Data.Ons
+{
+    public class OnsExcelReaderService(ILogger<OnsExcelReaderService> logger) : IOnsExcelReaderService
```
Comment (cofaulco) [inline src/SFA.DAS.PublicSectorOrganisations.Data/Ons/OnsExcelReaderService.cs]: any need to keep this class now?

**Review by cofaulco — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.PublicSectorOrganisations.Data/Ons/OnsExcelReaderService.cs (hunk 4 above)
Comment (Pauljgraham) [reply src/SFA.DAS.PublicSectorOrganisations.Data/Ons/OnsExcelReaderService.cs]: no removed

**Review by cofaulco — DISMISSED** [review]

**Review by cofaulco — APPROVED** [review]

**Review by furqanagwan — APPROVED** [review]

**Pull request merged**